### PR TITLE
fixed the spacing issue for books that have characters with multiple …

### PR DESCRIPTION
**Pull request by szarroug3**
…aliases separated by a comma; fixed some minor issues in parsing

### CODE REVIEW

File: lib/goodreads_parser.py [hunk 1]
```diff
@@ -289,9 +290,8 @@ def fullname_to_possible_aliases(self, fullname):
                     aliases.append("%s %s" % (title, surname))
             # Don't want the formats {ChristianName}, {Surname} and {ChristianName} {Lastname} in special cases
             # i.e. The Lord Ruler should never have "The Ruler", "Lord" or "Ruler" as aliases
-            if christian_name in self.COMMON_WORDS:
-                aliases.append('{0} {1}'.format(' '.join(parts), surname))
```
Comment (stoduk) [inline lib/goodreads_parser.py]: You've removed this alias (here parts has had christian and surname removed, so is whatever is left) - which might be correct but right now I can't see when we would have wanted it!  with "The Lord Ruler"  christian_name would be "The", surname "Ruler", so we would have added an alias of "Lord Ruler" - which seems wrong.

So maybe you've just removed something that was never needed, just wanted to check.

File: lib/goodreads_parser.py [hunk 2]
```diff
@@ -289,9 +290,8 @@ def fullname_to_possible_aliases(self, fullname):
...
-            else:
+            # Same for John the Great
+            if christian_name not in self.COMMON_WORDS and (len(parts) == 0 or parts[0] not in self.COMMON_WORDS):
```
Comment (stoduk) [inline lib/goodreads_parser.py]: General comment: this function seems very likely to get broken as we add more special cases, but also seems ripe for regression testing.  ie. we can throw a list of known names at it, and check it produces the right set of aliases.  Not something for now, obviously.

File: lib/book_parser.py [hunk 3]
```diff
@@ -191,7 +181,7 @@ def find_erl_and_encoding(self):
             nrecs, = unpack('>H', book_data[76:78])
             recs_start = 78 + (nrecs * 8) + 2
             erl, = unpack('>L', book_data[recs_start + 4:recs_start + 8])
-            codec = 'latin-1' if unpack('>L', book_data[recs_start + 28:recs_start + 32])[0] == 1252 else 'utf8'
+            codec = 'cp1252' if unpack('>L', book_data[recs_start + 28:recs_start + 32])[0] == 1252 else 'utf8'
```
Comment (stoduk) [inline lib/book_parser.py]: Not sure why you've made this, but if it works..

File: lib/book_parser.py [hunk 4]
```diff
@@ -106,19 +104,26 @@ def _get_paragraph_data(self, codec):
 
         # find all paragraphs (sections enclosed in html p tags) and their starting offset
         for node in re.finditer(PARAGRAPH_PAT, book_html):
-            # get plain text from paragraph and locations of letters from beginning of file
-            results = [(word.group(0)[1:-1].decode(codec), word.start(0))
-                       for word in re.finditer(PLAIN_TEXT_PAT, node.group(0))]
             word_loc = {'words': '', 'locs': [], 'char_sizes': []}
```
Comment (stoduk) [inline lib/book_parser.py]: What is this fixing?

I assume there is something going on here with length offsets where HTML tags are included..  Are you just trying to exclude any HTML tags from the word lists?

**Comment by stoduk** [comment]
You've committed a couple of test files - what are they for?  They seem to be copyrighted material, so that is probably not something you want to keep online.

If we want a real book for testing purposes then something out of copyright could do the job.  Though it might be just as useful to manufacture some test books - there won't be much benefit in having something book length when most the time we just want to test that particular corner cases work.

**Comment by szarroug3** [comment]
It was an accident actually. I was using them locally and forgot to remove
them before I committed.. Whoops! I'll remove them.

File: lib/book_parser.py (hunk 3 above)
Comment (szarroug3) [inline lib/book_parser.py]: just realized that i was using the wrong thing before according to some docs on mobi headers

File: lib/goodreads_parser.py [hunk 5]
```diff
@@ -205,8 +205,9 @@ def get_characters(self, entity_id):
                 desc = unicode(re.sub(r'\s+', ' ', desc[0]).strip().decode('utf-8').encode('latin-1'))
             else:
                 desc = u'No description found on Goodreads.'
-            alias_list = [re.sub(r'\s+', ' ', x).strip() for x in char_page.xpath('//div[@class="grey500BoxContent" and contains(.,"aliases")]/text()') if re.sub(r'\s+', ' ', x).strip()]
-            alias_list = [alias for aliases in alias_list for alias in aliases.split(',')]
+            alias_list = [x for x in char_page.xpath('//div[@class="grey500BoxContent" and contains(.,"aliases")]/text()')]
```
Comment (szarroug3) [inline lib/goodreads_parser.py]: will do

File: lib/goodreads_parser.py (hunk 1 above)
Comment (szarroug3) [inline lib/goodreads_parser.py]: yeah i realized it wasn't needed so i removed it

File: lib/book_parser.py (hunk 4 above)
Comment (szarroug3) [inline lib/book_parser.py]: i realized that when you looked in the book, it was messing up the paragraphs when you look at the places where our special list of names/words was listed.. it would print the wrong sections.. everything that was changed in this file was meant to fix that.. i found that when we parse the gunslinger by stephen king, it would parse incorrectly so i originally was trying to fix that issue but couldn't figure out what was there.. see #91 .. while i was working on that, i found this issue so i fixed it
